tests/utilities/discovery_service_process.py

```diff
@@ -0,0 +1,38 @@
+"""Generator class to create and terminate Discovery Service instance."""
+
+from ni_measurementlink_service._internal.discovery_client import (
+    _get_discovery_service_location,
+    _get_key_file_path,
+    _service_already_running,
+    _start_service,
+)
+
+
+class DiscoveryServiceProcess:
+    """Maintains the processes involved in creating and terminating discovery service."""
+
+    def __init__(self):
+        """Creates a DiscoveryServiceProcess instance."""
+        try:
+            self._proc = None
+            key_file_path = _get_key_file_path()
+            if _service_already_running(key_file_path):
+                return
+            self._proc = _start_service(_get_discovery_service_location(), key_file_path)
+        except Exception:
+            self._close_discovery_service()
+            raise
+
+    def __enter__(self):
```

Comment:
Add one of these (preferably the 1st, if it works):
- `from typing_extensions import Self`
- `Self = TypeVar("Self", bound=object)`

```suggestion
    def __enter__(self: Self) -> Self:
```